slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -100,12 +106,40 @@ public void export(final OutputStream output) {
     }
   }
 
+  @Override
+  public void registerGenesisValidatorsRoot(final Bytes genesisValidatorsRoot) {
+    jdbi.useTransaction(
+        SERIALIZABLE,
+        h -> {
+          final Optional<Bytes> dbGvr = metadataDao.findGenesisValidatorsRoot(h);
+          final Optional<Boolean> isValidGvr = dbGvr.map(gvr -> gvr.equals(genesisValidatorsRoot));
+          if (!isValidGvr.orElse(true)) {
+            throw new IllegalStateException(
+                String.format(
+                    "Genesis validators root %s does not match slashing protection db validators root",
```

Comment:
maybe "supplied genesis validators root %s does not match value recorded in database"?